src/main/java/com/agilethought/atsceapi/service/HistoryServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.agilethought.atsceapi.service;
+
+import com.agilethought.atsceapi.dto.NewHistoryRequest;
+import com.agilethought.atsceapi.dto.NewHistoryResponse;
+import com.agilethought.atsceapi.model.History;
+import com.agilethought.atsceapi.repository.HistoryRepository;
+import lombok.extern.slf4j.Slf4j;
+import ma.glasnost.orika.MapperFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class HistoryServiceImpl implements HistoryService{
+    @Autowired
+    private HistoryRepository historyRepository;
+
+    @Autowired
+    private MapperFacade orikaMapperFacade;
+
+    @Override
+    public NewHistoryResponse createHistory(NewHistoryRequest request) {
+        // Manual mapping from NewHistoryRequest to History classes.
+        History newHistory = new History(
+                null,
+                request.getType().byteValue(),
+                request.getUser_id(),
+                request.getFixedExpenses(),
+                request.getTotalHours(),
+                request.getTotalDays(),
+                request.getCostDay(),
+                request.getCostHour(),
+                request.getTaxIVA(),
+                request.getTaxISR_r(),
+                request.getTaxIVA_r(),
+                request.getTotal(),
+                request.getRevenue(),
+                request.getStatus() > 0
+        );
```

Comment:
You're missing "**project cost**" variable again.

Reply:
Change done, I have added the projectCost variable to the Model and the Service layers.